### PR TITLE
Query multiple tables

### PR DESCRIPTION
**Pull request by HunterX-7**
## Dear Code reviewer In this Project, I:

- Add the following features to the project:

### Vet Clinic project requirements

- Create a table named `owners` with the following columns:
  - `id`: integer (set it as autoincremented PRIMARY KEY)
  - `full_name`: string
  - `age`: integer
- Create a table named `species` with the following columns:
  - `id`: integer (set it as autoincremented PRIMARY KEY)
  - `name`: string
- Modify `animals` table:
  - Make sure that id is set as autoincremented PRIMARY KEY
  - Remove column `species`
  - Add column `species_id` which is a foreign key referencing `species` table
  - Add column `owner_id` which is a foreign key referencing the `owners` table

capture1

- Remember all this goes in `schema.sql` file.
- Insert the following data into the `owners` table:
  - Sam Smith 34 years old.
  - Jennifer Orwell 19 years old.
  - Bob 45 years old.
  - Melody Pond 77 years old.
  - Dean Winchester 14 years old.
  - Jodie Whittaker 38 years old.

capture2

- Insert the following data into the `species` table:
  - Pokemon
  - Digimon

capture3

- Modify your inserted animals so it includes the `species_id` value:
  - If the name ends in "mon" it will be Digimon
  - All other animals are Pokemon

capture4

- Modify your inserted animals to include owner information (`owner_id`):
  - Sam Smith owns Agumon.
  - Jennifer Orwell owns Gabumon and Pikachu.
  - Bob owns Devimon and Plantmon.
  - Melody Pond owns Charmander, Squirtle, and Blossom.
  - Dean Winchester owns Angemon and Boarmon.

capture5

- Remember these insertions and modifications happen in `data.sql` file.
- Write queries (using `JOIN`) to answer the following questions:
  - What animals belong to Melody Pond?
capture6
  - List of all animals that are pokemon (their type is Pokemon).
capture7
  - List all owners and their animals, remember to include those that don't own any animal.
capture8
  - How many animals are there per species?
capture9
  - List all Digimon owned by Jennifer Orwell.
capture10
  - List all animals owned by Dean Winchester that haven't tried to escape.
capture11
  - Who owns the most animals?
capture13
- Remember all these should be written in `queries.sql` file.
- **Take a screenshot of the results of your queries.**
- Remember to add the screenshot with the results of your queries to your Pull Request description.

## I Hope you have a great day and your support is much appreciated! 😄

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -56,4 +56,13 @@ select min(weight_kg) from animals where species = 'Pokemon';
...
+select avg(escape_attempts) from animals where species = 'Digimon' and date_of_birth > '1989-12-31' and date_of_birth < '2001-01-01';
+
+select animals.name from animals join owners on owners.id = animals.owners_id where owners.full_name = 'Melody Pond';
+select animals.name from animals join species on species.id = animals.species_id where species.name = 'Pokemon';
+select owners.full_name as owners, animals.name from animals full outer join owners on owners.id = animals.owners_id;
+select species.name as species, count(animals.name) from animals join species on species.id = animals.species_id group by species.name;
+select owners.full_name as owner, animals.name as name from animals join species on species.id = animals.species_id
+join owners on owners.id = animals.owners_id where owners.full_name = 'Jennifer Orwell' and species.id = 2;
+select animals.name from animals join owners on owners.id = animals.owners_id where animals.escape_attempts = 0 and owners.full_name = 'Dean Winchester';
+select owners.full_name as owners, count(animals.name) as count from animals join owners on owners.id = animals.owners_id group by owners.full_name;
```
Comment (Kamba56) [inline queries.sql]: - [x] Please kindly make this query return only one owner as the answer to who owns the most animals should not be more than one. You can arrange the output here in descending order and take only the first output using limit.

**Review by Kamba56 — CHANGES_REQUESTED** [review]
Hi @HunterX-7, 

Good job so far!!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [x] Great use of Github workflow!👏👏👏
- [x] All linters passed
- [x] Great job altering the animals table.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (HunterX-7) [reply queries.sql]: Thanks for the suggestion, the requested changes have been made, and thanks for your advice on the limit!

capture13

**Review by TadesseDev — APPROVED** [review]
### Hi @HunterX-7 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy (1)



## Highlights
 - [ ] Good job making the changes requested by the previous code reviewer. ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**